src/erlfmt_format.erl

```diff
@@ -653,7 +653,7 @@ clause_to_algebra({spec_clause, _Meta, Head, [Body], Guards}) ->
 guard_to_algebra(Guards, Separator) ->
     GuardsD = lists:map(fun expr_to_algebra/1, Guards),
     Doc = fold_doc(fun(GuardD, Acc) -> break(concat(GuardD, Separator), Acc) end, GuardsD),
-    group(Doc).
+    group(concat(maybe_force_breaks(has_any_break_between(Guards)), Doc)).
```

Comment:
Yes I think it would make sense to only do this for spec_clause
I tried this angle first, since doing it for spec_clause only would mean that we need to duplicate the path through expr_to_algebra for gaurd_or and gaurd_and
https://github.com/WhatsApp/erlfmt/blob/a3cb1769de6ac5952ba7a349e6a05f52ce7bd8a9/src/erlfmt_format.erl#L641
But I can definitely try that :)

Reply:
Yes, it definitely increases the complexity a bit in here. 

A workaround for that so that we don't need to duplicate the functions could be to add a special metadata to the guard nodes and later consume it in `guard_to_algebra` (passing in the additional `Meta` argument from `expr_to_algebra`).

Reply:
It ended up being less duplication than I initially thought, so I went with the duplication route.